ports/unix/coverage.c

```diff
@@ -650,4 +651,29 @@ STATIC mp_obj_t extra_coverage(void) {
 }
 MP_DEFINE_CONST_FUN_OBJ_0(extra_coverage_obj, extra_coverage);
 
+// function to run extra tests for things that can't be checked by scripts, and which have to destroy the GC heap
+STATIC mp_obj_t extra_coverage_gc(void) {
+    long heap_size = 1024;
+
+    for (unsigned j = 0; j < 256 * MP_BYTES_PER_OBJ_WORD; j++) {
+        char *heap = calloc(heap_size, 1);
+        gc_init(heap, heap + heap_size);
+
+        m_malloc(16);
+        void *o = gc_alloc(16, GC_ALLOC_FLAG_HAS_FINALISER);
+        for (int i = 0; i < heap_size / 16; i++) {
+            void *p = m_malloc_maybe(16);
+            if (!p) {
+                break;
+            }
+            *(void **)o = p;
+            o = p;
+        }
+        gc_collect();
+        heap_size++;
+    }
+    mp_printf(&mp_plat_print, "# end coverage.c gc test\n");
+    exit(0);
```

Comment:
This is quite brutal, to exit so abruptly!  I wonder if there's a better way to do it...

Reply:
I don't think this is a good way to do it because it prevents coverage/testing of the tear down and exiting of the executable.

Two options I can think of:
1. save and then restore the `mp_state_mem_t` struct in `mp_state_ctx_t` at the start/end of this function, which should restore the original heap on exit of this function
2. run this code at the start or end of main() (probably at the start, just before `gc_init()`).

Reply:
I did option (1) during merge.